core/src/test/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunnerTest.java

```diff
@@ -307,36 +316,54 @@ public void testDecoratePillarWithMountInfo() {
         when(cmExposedService.getServiceName()).thenReturn("CM");
         when(exposedServiceCollector.getClouderaManagerService()).thenReturn(cmExposedService);
 
-        Set<InstanceGroup> instanceGroups = new HashSet<>();
-        InstanceGroup workerInstanceGroup = new InstanceGroup();
         Template template = new Template();
         template.setTemporaryStorage(TemporaryStorage.EPHEMERAL_VOLUMES);
-        workerInstanceGroup.setTemplate(template);
-
-        Set<InstanceMetaData> workerInstanceMetaDatas = new HashSet<>();
-        InstanceMetaData worker1 = new InstanceMetaData();
-        worker1.setDiscoveryFQDN("fqdn1");
-        workerInstanceMetaDatas.add(worker1);
-        InstanceMetaData worker2 = new InstanceMetaData();
-        worker2.setDiscoveryFQDN(null);
-        workerInstanceMetaDatas.add(worker2);
-        workerInstanceGroup.setInstanceMetaData(workerInstanceMetaDatas);
-        instanceGroups.add(workerInstanceGroup);
-
-        InstanceGroup computeInstanceGroup = new InstanceGroup();
-        computeInstanceGroup.setTemplate(template);
-
-        Set<InstanceMetaData> computeInstanceMetaDatas = new HashSet<>();
-        InstanceMetaData compute1 = new InstanceMetaData();
-        compute1.setDiscoveryFQDN("fqdn2");
-        computeInstanceMetaDatas.add(compute1);
-        InstanceMetaData compute2 = new InstanceMetaData();
-        compute2.setDiscoveryFQDN(null);
-        computeInstanceMetaDatas.add(compute2);
-        computeInstanceGroup.setInstanceMetaData(computeInstanceMetaDatas);
-        instanceGroups.add(computeInstanceGroup);
+
+        Set<InstanceGroup> instanceGroups = new HashSet<>();
+        createInstanceGroup(template, instanceGroups, "fqdn1", null, "1.1.1.1", "1.1.1.2");
+        createInstanceGroup(template, instanceGroups, "fqdn2", null, "1.1.2.1", "1.1.2.2");
+        InstanceGroup gwIg = createInstanceGroup(template, instanceGroups, "gateway1", "gateway2", "1.1.3.1", "1.1.3.2");
+
+        when(stack.getPrimaryGatewayInstance()).thenReturn(
+                gwIg.getInstanceMetaDataSet().stream().filter(imd -> StringUtils.equals(imd.getDiscoveryFQDN(), "gateway1")).findFirst().get());
+        when(stackUtil.collectAndCheckReachableNodes(any(), any())).thenReturn(Sets.newHashSet(node("fqdn1"), node("fqdn2"), node("fqdn3"),
+                node("gateway1"), node("gateway3")));
 
         when(stack.getInstanceGroups()).thenReturn(instanceGroups);
-        underTest.runClusterServices(stack, cluster, Map.of());
+        underTest.runTargetedClusterServices(stack, cluster, Map.of("fqdn3", "1.1.1.1"));
```

Comment:
I would create a separate test method for the new public method `runTargetedClusterServices` because this way we will decrease the UT coverage on the original public method `runClusterServices`.
Could you please create a separate test method for it?

Also negative test cases for the new method would be nice to have even if the majority of the business logic is covered by the tests of `runClusterServices`.